aspnetcore/blazor/Security-guidance.md

```diff
@@ -0,0 +1,254 @@
+
+**For more in depth details about blazor server-side resource ussage we recommend you check the docummentation on reliability and scalability to get a better idea about expected resource consumption.**
+
+## Interactions with the browser (client)
+
+A client interacts with the server through JavaScript interop event dispatching and render completion. In the case of JS inteorp this communication goes both ways, from JavaScript to .NET and viceversa. Browser events are dispatched from client to server in an asynchronous fashion and the server responds asynchronously by producing new renders if necessary.
+
+### Invoking JavaScript functions from .NET
+Calls from .NET to the JavaScript need to watch for a few things:
+* All invocations have a configurable timeout after which they'll simply fail, returning a OperationCancelledException to the caller.
+  * There is a default timeout for the calls of around 1 minute.
+  * A cancellation token can be provided to control the cancellation on a per call basis.
+    * We recommend that you rely on the default call timeout where possible and that you time-bound any call to the client if you provide your own cancellation token.
+* The result of a JavaScript call can't be trusted.
+  * Our clients will simply search for the function to invoke, invoke it and produce either the result or an error. A malicious client, could however try to return an error to cause an issue in the application or an unexpected result to try induce some unintended behavior in the server.
+    * We recommend that you wrap any JS interop call within a try-catch block to account for any error that might happen during the invocation.
+    * We recommend that you validate any data comming out from a JS interop invocation before taking any action. This includes error messages.
+
+### Exposing .NET methods to the browser.
+Calls from JavaScript to .NET are not to be trusted. When you expose a .NET function to JavaScript you need to consider how that function might be invoked.
+* We recommend that you treat any .NET method exposed to JavaScript as you would a public endpoint in your app.
+  * Validate the input.
+    * Ensure tha the values are whithin the expected ranges.
+    * Ensure that the user has permission to perform the action he is requesting.      
+  * Make sure you don't allocate an excessive amount of resources as part of the invocation (CPU, Memory, etc).
+  * Take into account that static and instance methods can be exposed to JavaScript clients and that you should take special considerations to avoid sharing state between sessions.
+    * For instance methods exposed through the use of DotNetReference objects, if originally created through DI should be registered as scoped objects (this applies to any DI service your blazor server-side app uses).
+    * For static methods you should avoid having any sort of state that can't be scoped to the client.
```

Comment:
I think it's more a matter of being *aware* that for static methods, all users on the same server will be sharing the same state. That's not usually desirable, but occasionally is.